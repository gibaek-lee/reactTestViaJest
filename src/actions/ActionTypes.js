/* Declare action types */

//of App.js
export const INSERT = 'INSERT';

//of Counter.js
export const INCREASE = 'INCREASE';
export const DECREASE = 'DECREASE';

//of NameForm.js
export const SUBMIT = 'SUBMIT';
export const CHANGE = 'CHANGE';
