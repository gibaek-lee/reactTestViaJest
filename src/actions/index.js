/* Declare action creators */

import * as types from './ActionTypes';

export const increase = () => ({
  type: types.INCREASE
});

export const decrease = () => ({
  type: types.DECREASE
});

export const change = (name) => ({//store의 action parameter(action.name)
  type: types.CHANGE,
  name: name
});

export const insert = () => ({
  type: types.INSERT
});

export const submit = () => ({
  type: types.SUBMIT
});
