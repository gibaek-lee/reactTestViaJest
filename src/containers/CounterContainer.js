import Counter from '../components/Counter';
import * as actions from '../actions';
import { connect } from 'react-redux';

const mapStateToProps = (state) => ({
  value: state.valueData.value
});

const mapDispatchToProps = (dispatch) => ({
  onIncrease: () => dispatch(actions.increase()),
  onDecrease: () => dispatch(actions.decrease())
});

const CounterContainer = connect(
  mapStateToProps, mapDispatchToProps
)(Counter);

export default CounterContainer;
