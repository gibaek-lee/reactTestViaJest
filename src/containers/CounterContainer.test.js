import React from 'react';
import { mount } from 'enzyme';//enzyme full rendering
import CounterContainer from './CounterContainer';
import configureMockStore from 'redux-mock-store';
import * as valueActions from '../actions/index';

describe('CounterContainer', () => {
  let component = null;
  const mockStore = configureMockStore();

  let store = mockStore({
    valueData: {
      value: 1
    }
  });

  it('renders properly', () => {
    component = mount(<CounterContainer store={store}/>)
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('dispatches INCREASE action', () => {
    component.find('button').at(0).simulate('click');
    //store.getActions() 스토어에 dispatch된 action들의 목록
    expect(store.getActions()[0]).toEqual(valueActions.increase());
  });

  it('dispatches DECREASE action', () => {
    component.find('button').at(1).simulate('click');
    expect(store.getActions()[1]).toEqual(valueActions.decrease());
    //console.log(store.getActions());
  });
});
