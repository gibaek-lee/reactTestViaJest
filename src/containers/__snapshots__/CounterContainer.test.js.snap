// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterContainer matches snapshot 1`] = `
<Connect(Counter)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Counter
    onDecrease={[Function]}
    onIncrease={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    value={1}
  >
    <div>
      <h1>
        Counter
      </h1>
      <h2>
        1
      </h2>
      <button
        onClick={[Function]}
      >
        +
      </button>
      <button
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </Counter>
</Connect(Counter)>
`;
